### ISSUE TITLE
Unnecessary iterations on successful payment processing

### ISSUE THREAD

**Issue by JudeNiroshan** [commercetools/commercetools-payone-integration#312]
## Problem :feelsgood:

We found out while debugging this particular line has been called for 20 times. ⚠️ It is not necessary to loop during successful payment handling. Iterations are intended to retry for failed scenarios; not for successful conditions. 

https://github.com/commercetools/commercetools-payone-integration/blob/69c6d0ac0c85aeddcf6982341bf12ac07d87cb4e/service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java#L58

```
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
Breakpoint reached at com.commercetools.pspadapter.payone.PaymentHandler.handlePayment(PaymentHandler.java:59)
```

## Solution :octocat:

Do not loop for 20 times when everything works without a problem. Break from the loop.

**Comment by butenkor** [comment]
Could you also point exactly to a place where this happens? The idea of the loop is to retry on errors up to 20 times. Maybe while debugging you had a retriable error on every run?

**Comment by lojzatran** [comment]
> Could you also point exactly to a place where this happens? The idea of the loop is to retry on errors up to 20 times. Maybe while debugging you had a retriable error on every run?

I updated the task.

**Comment by JudeNiroshan** [comment]
Fixed by #323
